### DISCUSSION TITLE
Getting "Unknown file extension .scss" when component uses Element Plus

### DISCUSSION THREAD

**Discussion by LondonAppDev** [vitest-dev/vitest#2609] [Q&A] [answered]
I'm using Vitest to test a component that uses Element Plus. When I use the Element Plus plugin in vite.conf.ts, I get the following error if the component I'm testing uses any Element Plus component:

```
TypeError: Unknown file extension ".scss" for /home/projects/vitejs-vite-zcdxhn/node_modules/element-plus/theme-chalk/src/button.scss
```

I have created a StackBlitz to replicate the issue here: https://stackblitz.com/edit/vitejs-vite-avqnvz?file=vite.config.js

My vite.conf.js looks like this:

```js
import { defineConfig } from 'vite';
import vue from '@vitejs/plugin-vue';
import ElementPlus from 'unplugin-element-plus/vite';

export default defineConfig({
  plugins: [vue(), ElementPlus({ useSource: true })],
});

```

If I remove the `ElementPlus` plugin and import the Element Plus styles globally in `main.ts`, the issue goes away. So the problem seems to be related specifically to using that plugin.

Is there some kind of configuration I am missing which would make Vitest work?

Thanks in advance...

**Comment by sheremet-va** [comment, marked as answer]
You should inline dependencies that don't bundle CSS files:

```ts
// vitest.config.ts
export default {
  test: {
    deps: {
      inline: ['element-plus']
    }
  }
}
```

**Reply by LondonAppDev** [reply]
Thanks so much, this worked. 

Would you mind explaining what `inlined` does? 

The docs state:

 > Vite will process inlined modules. This could be helpful to handle packages that ship .js in ESM format (that Node can't handle).

But it isn't clear why this would be needed for a package that uses scss/css?

**Reply by sheremet-va** [reply]
Non-inlined (external) packages are processed by Node, and not Vite. All node_modules packages are externalized by default for performance reasons. Node doesn't understand what it means to import CSS file, so we should process file with Vite before Node can do something with it.

**Reply by LondonAppDev** [reply]
Excellent. Makes sense. Thanks so much 🙏
